crates/pallet-executor/src/lib.rs

```diff
@@ -671,11 +675,13 @@ impl<T: Config> Pallet<T> {
 
         // Remove the expired receipts once the receipts cache is full.
         if let Some(to_prune) = primary_number.checked_sub(&T::ReceiptsPruningDepth::get()) {
-            let block_hash = BlockHash::<T>::take(to_prune);
-            for receipt_hash in <ReceiptVotes<T>>::iter_key_prefix(block_hash) {
-                Receipts::<T>::remove(receipt_hash);
-            }
-            let _ = <ReceiptVotes<T>>::clear_prefix(block_hash, u32::MAX, None);
+            BlockHash::<T>::mutate_exists(to_prune, |maybe_block_hash| {
```

Comment:
This is more correct, but my comment was actually about `BlockHash` itself. I don't think it should be defined with `ValueQuery`, it must be `OptionQuery`

Reply:
Why though? It's a storage item totally internal to the system. Unless it's obviously error-prone, using `ValueQuery` is generally simpler than `OptionQuery` regarding the code itself.

Reply:
Just feels wrong to me to use it here. You will always get something even if it is not there and isn't even necessarily supposed to be and there isn't any meaningful "default" value in this case either. I can imagine this causing tricky issues down the line.

Reply:
I see your concern. I did though about this point, we always validate these unsigned extrinsics before applying them, which means the block hash can be ensured to be a meaningful value, the default value won't pass the validation. Considering the convenience it brings, I chose to use the `ValueQuery`.